tests/unit/customizations/ecs/test_monitorexpressgatewayservice.py

```
@@ -77,44 +77,265 @@ def test_non_monitoring_error_bubbles_up(self, mock_isatty):
+            region="us-west-2", endpoint_url=None, verify_ssl=True
+        )
+
+        command._run_main(parsed_args, parsed_globals)
 
 
 class TestECSExpressGatewayServiceWatcher:
     """Test the watcher class through public interface"""
 
-    @patch('sys.stdout.isatty')
-    def test_is_monitoring_available_with_tty(self, mock_isatty):
-        """Test is_monitoring_available returns True when TTY is available"""
-        mock_isatty.return_value = True
+    def setup_method(self):
+        self.app_session = create_app_session(output=DummyOutput())
+        self.app_session.__enter__()
+        self.mock_client = Mock()
+        self.service_arn = (
+            "arn:aws:ecs:us-west-2:123456789012:service/my-cluster/my-service"
+        )
+
+    def teardown_method(self):
+        if hasattr(self, 'app_session'):
+            self.app_session.__exit__(None, None, None)
+
+    def _create_watcher_with_mocks(
+        self, resource_view="RESOURCE", timeout=1, display_mode="interactive"
+    ):
+        """Helper to create watcher with mocked display strategy"""
+        mock_display_strategy = Mock()
+
+        # Create watcher with mocked strategy
+        watcher = ECSExpressGatewayServiceWatcher(
+            self.mock_client,
+            self.service_arn,
+            resource_view,
+            display_mode,
+            timeout_minutes=timeout,
+            display_strategy=mock_display_strategy,
+        )
+
+        # Mock exec to call the collector once and print output
+        collector = watcher.collector
+
+        def mock_exec():
+            try:
+                output = collector.get_current_view("⠋")
+                print(output)
+                print("Monitoring Complete!")
+            except Exception as e:
+                # Re-raise expected exceptions
+                if isinstance(e, (ClientError, MonitoringError)):
+                    raise
+                # For other exceptions, just print and complete
```

Not sure how I feel about swallowing unexpected errors. Can we discuss raising errors somehow and having a graceful error-raise mode?